internal/aggregation/aggregator_impl.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregation
+
+import (
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+type aggregator struct {
+	id int32
+}
+
+func (ag *aggregator) WriteData(output serialization.DataOutput) (err error) {
+	panic("implement me")
+}
+
+func (ag *aggregator) ReadData(input serialization.DataInput) (err error) {
+	panic("implement me")
+}
+
+func (*aggregator) FactoryID() int32 {
+	return FactoryID
+}
+
+func (ag *aggregator) ClassID() int32 {
+	return ag.id
+}
+
+func newAggregator(id int32) *aggregator {
+	return &aggregator{id: id}
+}
+
+type Count struct {
+	*aggregator
+	attributePath string
+}
+
+func NewCount(attributePath string) *Count {
+	return &Count{aggregator: newAggregator(count), attributePath: attributePath}
+}
+
+func (c *Count) ReadData(input serialization.DataInput) (err error) {
+	c.attributePath, err = input.ReadUTF()
+	// member side field, not used in client
```

Comment:
You should create new aggregator. Also, serialization should be tested like `Predicate` and `Projection`.